src/services/external/excalidraw-backend/utils/handlers.ts

```diff
@@ -125,7 +127,12 @@ export const serverBroadcastEventHandler = (
   socket: SocketIoSocket
 ) => {
   socket.broadcast.to(roomID).emit(CLIENT_BROADCAST, data);
-  socket.data.lastContributed = Date.now();
+
+  const decodedData = new TextDecoder().decode(data);
+  const decryptedData = JSON.parse(decodedData);
+  if (decryptedData.type === ClientBroadcastPayloadType.SCENE_UPDATE) {
+    socket.data.lastContributed = Date.now();
```

Comment:
if an event is going through this listener, then it's a contibution - you have compeletely changed the logic.